google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/readrows/ReadRowsRetryTest.java

```diff
@@ -124,6 +134,98 @@ public void immediateRetryTest() {
     Truth.assertThat(actualResults).containsExactly("k1", "r1", "r2").inOrder();
   }
 
+  public ApiException largeRowExceptionWithTrailers(String rowKey) {
+    ErrorInfo errorInfo =
+        ErrorInfo.newBuilder()
+            .setReason("LargeRowReadError")
+            .setDomain("bigtable.googleapis.com")
+            .putMetadata("rowKey", rowKey)
+            .build();
+
+    Any packedErrorInfo = Any.pack(errorInfo);
+    // ErrorDetails errorDetails = ErrorDetails.builder()
+    //     .setRawErrorMessages(Collections.singletonList(packedErrorInfo))
+    //     .build();
+
+    ErrorDetails errorDetails =
+        ErrorDetails.builder().setRawErrorMessages(ImmutableList.of(packedErrorInfo)).build();
+
+    Metadata trailers = new Metadata();
+    byte[] status =
+        com.google.rpc.Status.newBuilder().addDetails(Any.pack(errorInfo)).build().toByteArray();
+    trailers.put(ERROR_DETAILS_KEY, status);
+    return (ApiException)
+        (new UnavailableException(
+            new StatusRuntimeException(Status.FAILED_PRECONDITION, trailers),
+            GrpcStatusCode.of(Code.FAILED_PRECONDITION),
+            false,
+            errorDetails));
+  }
+
+  /**
+   * AdditionalTests - 1. only query for large rows - empty response? 2. 1st row is large row - then
```

Comment:
instead of the comment the test itself should describe what the test is doing. for example largeRowTestBasic could be largeRowRangesTest, and add a comment only describe what this test is testing